lib/ocpp/v201/charge_point.cpp

```diff
@@ -1801,7 +1802,14 @@ void ChargePoint::sign_certificate_req(const ocpp::CertificateSigningUseEnum& ce
 
     const auto csr = this->evse_security->generate_certificate_signing_request(
         certificate_signing_use, country.value(), organization.value(), common.value(), should_use_tpm);
-    req.csr = csr;
+
+    if (!csr.has_value()) {
+        EVLOG_error << "CSR generation was unsuccessful for sign request: "
```

Comment:
In this case its reasonable to send a SecurityEventNotification.req to the CSMS. There is no predefined type in this appendix provided by the spec, so I would use something like `CSRGenerationFailed`